src/test/java/org/openmrs/contrib/qaframework/automation/RolesAndPrivilegesSteps.java

```diff
@@ -0,0 +1,105 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.page.AddEditNewPrivilegePage;
+import org.openmrs.contrib.qaframework.page.AddEditNewRolePage;
+import org.openmrs.contrib.qaframework.page.ConfigureMetadataPage;
+import org.openmrs.contrib.qaframework.page.ManagePrivilegesPage;
+import org.openmrs.contrib.qaframework.page.ManageRolesPage;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+public class RolesAndPrivilegesSteps extends Steps {
+	private static final String PRIVILEGE_NAME_TO_BE_CREATED_AND_EDITED = "PrivilegeTest";
+	private static final String ROLE_NAME_TO_BE_CREATED_AND_EDITED = "RoleTest";
+	private static final String ROLE_DESCRIPTION_TO_BE_CREATED = "for e2e automation test";
+	private static final String PRIVILEGE_DESCRIPTION_TO_BE_CREATED = "This privilege is just one developed for the roles&privileges e2e test";
+	private ManagePrivilegesPage manageprivilegesPage;
+	private ConfigureMetadataPage configuremetadatapage;
+	private ManageRolesPage manageRolesPage;
+	private AddEditNewPrivilegePage addNewPrivilegePage;
+	private AddEditNewRolePage addNewRolePage;
+	
+	@Before(RunTest.HOOK.SELENIUM_ROLES_AND_PRIVILEGES)
+	public void visitDashboard(){
+		initiateWithLogin();
+	}
+	
+    @After(RunTest.HOOK.SELENIUM_ROLES_AND_PRIVILEGES)
+    public void destroy() {
+        quit();
+    }
+    
+    @Given("User clicks on configure metadata link from home page")
+    public void launchHomeconfigureMetaData(){
+    	configuremetadatapage = homePage.goToConfigureMetadata();
```

Comment:
Am not sure if you have tried out my suggestion of using administrationPage.clickOnManageRoles which is working perfect instead of using configuremetadata where there is a bug to fox https://qa-refapp.openmrs.org/openmrs/admin/users/role.list . i would think this would keep you going

Reply:
thanks for suggestion @sherrif10 but i find it better to wait for the ticket logged for the bug above (by ignoring that last scenario of the **rolesAndPrivileges** test workflow) than diverting to a whole new dashboard (admininstration) into the system administration hence diverting from the ticket description